packages/ember-engines/tests/unit/engine-instance-test.js

```diff
@@ -157,3 +158,77 @@ test('it can build a child engine instance with dependencies that are aliased',
     );
   });
 });
+
+test('it deprecates camelized engine names', async function(assert) {
+  assert.expect(5);
+
+  let deprecationMessages = [];
+
+  registerDeprecationHandler((message, options, next) => {
```

Comment:
Not sure if this is the correct way. I also tried using `ember-qunit-assert-helpers` but asserting that no deprecation was thrown didn't work for me. The `globals-resolver` deprecation was also thrown during this test and I couldn't manage to filter it out.

Reply:
I'd do this setup instead:


```js
// packages/ember-engines/tests/test-helper.js

let deprecations;
registerDeprecationHandler((message, options, next) => {
  // in case a deprecation is issued before a test is started
  if (!deprecations) {
    deprecations = [];
  }

  deprecations.push(message);
  next(message, options);
});

QUnit.testStart(function () {
  deprecations = [];
});

QUnit.assert.noDeprecations = function (callback) {
  let originalDeprecations = deprecations;
  deprecations = [];

  callback();
  this.deepEqual(deprecations, [], 'Expected no deprecations during test.');

  deprecations = originalDeprecations;
};

QUnit.assert.deprecations = function (callback, expectedDeprecations) {
  let originalDeprecations = deprecations;
  deprecations = [];

  callback();
  this.deepEqual(deprecations, expectedDeprecations, 'Expected deprecations during test.');

  deprecations = originalDeprecations;
};

QUnit.assert.deprecationsInclude = function (expected) {
  this.pushResult({
    result: deprecations.indexOf(expected) > -1,
    actual: deprecations,
    message: `expected to find \`${expected}\` deprecation`,
  });
};
```

Which is basically what I do over in https://github.com/emberjs/ember-test-helpers/blob/d1a05f4409b58c1fd1689a3afbabcf8f8e4047ab/tests/test-helper.js

Reply:
The issue with the current approach is that the test essentially mutates global state (there is no way to _undo_ the registerDeprecationHandler).